### PR TITLE
My account integration

### PR DESCRIPTION
**Pull request by baomastr**

### CODE REVIEW

File: src/components/Dashboard/BorrowLimit.tsx [hunk 1]
```diff
@@ -29,23 +27,7 @@ const CardWrapper = styled.div`
...
-      setAvailable(total.dp(2, 1).toString(10));
-      setBorrowPercent(
-        total.isZero() || total.isNaN()
-          ? 0
-          : userTotalBorrowBalance.div(total).times(100).dp(0, 1).toNumber(),
-      );
-    }
-  }, [userTotalBorrowBalance, userTotalBorrowLimit]);
-
+  const { available, borrowPercent } = useBorrowLimit();
```
Comment (baomastr) [inline src/components/Dashboard/BorrowLimit.tsx:28]: just moved current calculations into hooks in order to reuse it before the logics refactoring

File: src/components/Dashboard/WalletBalance.tsx [hunk 2]
```diff
@@ -80,123 +73,11 @@ const BalancerWrapper = styled.div`
...
-        withXVS,
-      });
-    }
-    return () => {
-      isMounted = false;
-    };
-  }, [withXVS]);
-
-  const formatValue = (value: $TSFixMe) => `$${format(getBigNumber(value).dp(2, 1).toString(10))}`;
+  const { totalSupply, formatValue, netAPY, withXVS, setWithXVS, totalBorrow } = useWalletBalance({
```
Comment (baomastr) [inline src/components/Dashboard/WalletBalance.tsx:77]: as well as `useWalletBalance`

File: src/components/Dashboard/useWalletBalance.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import { useVaiVault } from '../../hooks/useContract';
+import { getBigNumber } from '../../utilities/common';
+import { Setting } from '../../types';
+
+export interface WalletBalanceProps {
+  settings: Setting;
+  setSetting: (setting: Partial<Setting> | undefined) => void;
+}
+
+const format = commaNumber.bindWith(',', '.');
```
Comment (baomastr) [inline src/components/Dashboard/useWalletBalance.ts:16]: this part should be fixed after branch update

File: src/components/v2/MyAccount/index.tsx [hunk 4]
```diff
@@ -37,10 +40,11 @@ export const MyAccount = ({
   borrowLimitSum,
   safeLimit,
   sliderProps,
+  onSwitch,
+  isSwitched,
 }: IMyAccountProps) => {
-  const [isToggleSwitched, setToggleSwitched] = useState(true);
```
Comment (baomastr) [inline src/components/v2/MyAccount/index.tsx]: this state is also handled by parent component.

**Comment by therealemjy** [comment]
Hey man,
We should use React Query for all our requests. If you want to check some examples you can have a look at the repo I put in place several weeks ago as a spike: https://github.com/VenusProtocol/spike-venus-protocole-interface-state-management

**Comment by baomastr** [comment]
> Hey man,
> We should use React Query for all our requests. If you want to check some examples you can have a look at the repo I put in place several weeks ago as a spike: https://github.com/VenusProtocol/spike-venus-protocole-interface-state-management

got 404

**Comment by baomastr** [comment]
> Hey man,
We should use React Query for all our requests. If you want to check some examples you can have a look at the repo I put in place several weeks ago as a spike: https://github.com/VenusProtocol/spike-venus-protocole-interface-state-management

it gonna be used in next PR as a data layer refactoring issue.
current realisation works on existing data calculations

File: src/components/Dashboard/WalletBalance.tsx [hunk 5]
```diff
@@ -1,21 +1,15 @@
-import React, { useEffect, useState, useCallback } from 'react';
+import React from 'react';
```
Comment (coreyar) [inline src/components/Dashboard/WalletBalance.tsx:1]: We can move this file to `pages/Dashboard`

**Comment by baomastr** [comment]
got the new PR: https://github.com/VenusProtocol/venus-protocol-interface/pull/272

**Pull request closed without merging**